compose/src/androidTest/java/com.bottlerocketstudios.compose/auth/AuthCodeScreenTest.kt

```diff
@@ -0,0 +1,131 @@
+        val authCodeScreenImage = composeTestRule.onNodeWithContentDescription(authCodeScreenImageContentDescription)
+        authCodeScreenImage.assertIsDisplayed()
+    }
+
+    @Test
+    fun devOptionsButton_doesNotExist() {
+        // State
+        val state = AuthCodeState(
+            requestUrl = "".asMutableState(),
+            devOptionsEnabled = false,
+            onAuthCode = {},
+            onLoginClicked = {},
+            onSignupClicked = {},
+            onDevOptionsClicked = {},
+            showToolbar = {}
+        )
+
+        composeTestRule.setContent {
+            ArchitectureDemoTheme {
+                AuthCodeScreen(state, webViewNavigator)
+            }
+        }
+
+        val devOptionsButtonText = composeTestRule.activity.resources.getString(R.string.dev_options_button).uppercase()
+        // retrieve compose button instance using button text as identifier
+        val devOptionsButton = composeTestRule.onNodeWithText(devOptionsButtonText)
+        devOptionsButton.assertDoesNotExist()
+    }
+
+    @Test
+    fun devOptionsButton_doesExist_andIsDisplayed() {
+        // State
+        val state = AuthCodeState(
+            requestUrl = "".asMutableState(),
+            devOptionsEnabled = true,
+            onAuthCode = {},
+            onLoginClicked = {},
+            onSignupClicked = {},
+            onDevOptionsClicked = {},
+            showToolbar = {}
+        )
+
+        composeTestRule.setContent {
+            ArchitectureDemoTheme {
+                AuthCodeScreen(state, webViewNavigator)
+            }
+        }
+
+        val devOptionsButtonText = composeTestRule.activity.resources.getString(R.string.dev_options_button).uppercase()
+        // retrieve compose button instance using button text as identifier
+        val loginButton = composeTestRule.onNodeWithText(devOptionsButtonText)
+        loginButton.assertIsDisplayed()
+        loginButton.assertExists()
+    }
+
+    @Test
+    fun loginButton_isClicked() {
+        // State
+        var onLoginWasClicked = false
+        val state = AuthCodeState(
+            requestUrl = "".asMutableState(),
+            devOptionsEnabled = true,
+            onAuthCode = {},
+            onLoginClicked = { onLoginWasClicked = true },
+            onSignupClicked = {},
+            onDevOptionsClicked = {},
+            showToolbar = {}
+        )
+
+        composeTestRule.setContent {
+            ArchitectureDemoTheme {
+                AuthCodeScreen(state, webViewNavigator)
+            }
+        }
+
+        val loginButtonText = composeTestRule.activity.resources.getString(R.string.login_button).uppercase()
+        // retrieve compose button instance using button text as identifier
+        val loginButton = composeTestRule.onNodeWithText(loginButtonText)
+        loginButton.performClick()
+        loginButton.assertTextEquals(loginButtonText)
+        assert(onLoginWasClicked)
```

Comment:
@IvanTelisman I'd prefer to use `Truth` assertions here if possible:

```suggestion
        assertThat(onLoginWasClicked).isTrue()
```

_(Note that you'll need to import `assertThat` and can't just use the GitHub suggestion to commit from the PR)_

Reply:
Ah yes good ol Truth, good call makes those assertions easier to read too. Just a heads-up I had to create an additional variation of the truth dependency as an androidTestImplementation so that it would work as an androidTest. Got it added, thanks @br-jared-stockton !

Reply:
Thanks for adding this!